File: src/test/java/org/example/cool/TestWithDrivers.java
package org.example.cool;

import org.example.Utils;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.openqa.selenium.WebDriver;

import java.util.List;

public abstract class TestWithDrivers {
    public List<WebDriver> drivers;

    @BeforeAll
    public static void prepareDrivers() {
        Utils.prepareDrivers();
    }

    @BeforeEach
    private void beforeEach() { // todo before all?
        drivers = Utils.getDrivers();
    }

    @AfterEach
    private void afterEach() {
        drivers.forEach(WebDriver::quit);
    }
}

File: src/test/java/lab3/cool/TestWithDrivers.kt
package lab3.cool

import lab3.Utils
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.BeforeEach
import org.openqa.selenium.WebDriver
import java.util.function.Consumer

abstract class TestWithDrivers {
    var drivers: List<WebDriver>? = null
    @BeforeEach
    private fun beforeEach() { // todo before all?
        drivers = Utils.drivers
    }

    @AfterEach
    private fun afterEach() {
        drivers!!.forEach(Consumer { obj: WebDriver -> obj.quit() })
    }

    companion object {
        @BeforeAll
        fun prepareDrivers() {
            Utils.prepareDrivers()
        }
    }
}